crates/pathfinder/src/state/sync.rs

```diff
@@ -205,12 +205,10 @@ where
                     match &mut *state.status.write().await {
                         SyncStatus::False(_) => {}
                         SyncStatus::Status(status) => {
-                            status.current_block_hash = block_hash;
-                            status.current_block_num = StarknetBlockNumber(block_num);
+                            status.current = NumberedBlock::from((block_hash, StarknetBlockNumber(block_num)));
 
-                            if status.highest_block_num.0 <= block_num {
-                                status.highest_block_num = StarknetBlockNumber(block_num);
-                                status.highest_block_hash = block_hash;
+                            if status.highest.number.0 <= block_num {
+                                status.highest = status.current;
```

Comment:
I was in #296 commenting that this could just be a max of the two, but NumberedBlock probably shouldn't have an Ord or PartialOrd implementation based on the number after all, since we cannot say anything about the hash. Should only derive be used, the resulting bug coming with slice sorting would be quite difficult and rare thing to debug.

Reply:
Yes, I briefly thought about how updating should work when numbers are equal, but hashes differ. And resulted in saying the newer one should always win which meant `max` wouldn't work.